include/boost/gil/threshold.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef BOOST_GIL_THRESHOLD_HPP
+#define BOOST_GIL_THRESHOLD_HPP
+
+template <typename SrcView, typename DstView,typename typ>
+void bin_thresh(SrcView const& src, DstView const& dst,typ max,typ min,typ thresh)
```

Comment:
In the description of your PR you state that you are proposing a sample application, but here you are adding new feature to the library itself. I'm afraid it is not acceptable.

Please, move this code to the `threshold_demo.cpp` app.